src/main/java/com/renatusnetwork/momentum/gameplay/listeners/JoinLeaveListener.java

```diff
@@ -186,6 +189,18 @@ public void onQuit(PlayerQuitEvent event)
             playerStats.resetLevel();
         }
 
+        Squad squad = playerStats.getSquad();
+        if (playerStats.getSquad() != null) {
+            if (SquadManager.isLeader(playerStats)) {
+                SquadManager.notifyMembers(squad, "&3The squad has been disbanded because the leader went offline");
+                squadManager.disband(squad);
+            }
+            else {
+                squadManager.leave(playerStats, false);
+                SquadManager.notifyMembers(squad, "&9SqC &3" + playerStats.getDisplayName() + " &bhas left the squad");
```

Comment:
I wonder if we should have a system in place where if a player disconnects, they have a few minutes before they're auto kicked. Just in case of relogging or accidentally disconnected. Thoughts? Maybe a minute or two?

Reply:
should we also transfer leadership randomly to someone else if the leader leaves or keep it how it is with the squad disbanding if the leader leaves?

Reply:
Yeah, if the leader leaves and the timer goes out it should randomly select the new leader, or choose the next party member that has been there the longest which would require a timestamp of each player. Not too hard considering it's just adding to cache and not db and that would be preferred if possible